qcodes/dataset/plotting.py

```diff
@@ -229,3 +240,118 @@ def plot_on_a_plain_grid(x: np.ndarray, y: np.ndarray,
     else:
         colorbar = ax.figure.colorbar(colormesh, ax=ax)
     return ax, colorbar
+
+
+_SI_UNITS = {'m', 's', 'A', 'K', 'mol', 'rad', 'Hz', 'N', 'Pa', 'J',
+             'W', 'C', 'V', 'F', 'ohm', 'Ohm',
+             '\N{GREEK CAPITAL LETTER OMEGA}', 'S', 'Wb', 'T', 'H'}
+
+_ENGINEERING_PREFIXES = OrderedDict({
+    -24: "y",
+    -21: "z",
+    -18: "a",
+    -15: "f",
+    -12: "p",
+     -9: "n",
+     -6: "\N{GREEK SMALL LETTER MU}",
+     -3: "m",
+      0: "",
+      3: "k",
+      6: "M",
+      9: "G",
+     12: "T",
+     15: "P",
+     18: "E",
+     21: "Z",
+     24: "Y"
+})
+
+_THRESHOLDS = OrderedDict(
+    {10**(scale + 3): scale for scale in _ENGINEERING_PREFIXES.keys()})
+
+
+def _scale_formatter(tick_value, pos, factor):
+    """
+    Function for matplotlib.ticker.FuncFormatter that scales the tick values
+    according to the given `scale` value.
+    """
+    return "{0:g}".format(tick_value*factor)
+
+
+def _make_rescaled_ticks_and_units(data_dict):
+    """
+    Create a ticks formatter and a new label for the data that is to be used
+    on the axes where the data is plotted.
+
+    ...
+
+    Args:
+        data_dict: a dictionary of the following structure
+            {
+                'data': <1D numpy array of points>,
+                'name': <name of the parameter>,
+                'label': <label of the parameter or ''>,
+                'unit': <unit of the parameter or ''>
+            }
+
+    Returns:
+        a tuple with the ticks formatter and the new label; in case it is not
+        possible to scale, the returned values are None's
+    """
+    ticks_formatter = None
+    new_label = None
+
+    unit = data_dict['unit']
+
+    if unit in _SI_UNITS:
+        maxval = np.nanmax(data_dict['data'])
```

Comment:
Should this be the max of the absolute value?

Reply:
i think it should. but the original implementation did not have it :) I'll change it and add a test for it.